tools/createNewRollup/createNewRollup.ts

```diff
@@ -0,0 +1,399 @@
+    ) as PolygonRollupManager;
+
+    // Load global exit root manager
+    const globalExitRootManagerFactory = await ethers.getContractFactory("PolygonZkEVMGlobalExitRootV2", deployer);
+    const globalExitRootManagerAddress = await rollupManagerContract.globalExitRootManager();
+    const globalExitRootManagerContract = globalExitRootManagerFactory.attach(
+        globalExitRootManagerAddress
+    ) as PolygonRollupManager;
+
+    // Check if the deployer has right to deploy new rollups from rollupManager contract
+    const DEFAULT_ADMIN_ROLE = ethers.ZeroHash;
+    if ((await rollupManagerContract.hasRole(DEFAULT_ADMIN_ROLE, deployer.address)) == false) {
+        throw new Error(
+            `Deployer does not have admin role. Use the test flag on deploy_parameters if this is a test deployment`
+        );
+    }
+
+    // Check chainID
+    let rollupID = await rollupManagerContract.chainIDToRollupID(chainID);
+    if (Number(rollupID) !== 0) {
+        throw new Error(`Rollup with chainID ${chainID} already exists`);
+    }
+    // Check rollupTypeId
+    const rollupType = await rollupManagerContract.rollupTypeMap(createRollupParameters.rollupTypeId);
+    const consensusContractAddress = rollupType[0];
+    const verifierType = Number(rollupType[3]);
+    if (consensusContractName === "PolygonPessimisticConsensus" && verifierType !== 1) {
+        throw new Error(`Verifier type should be 1 for ${consensusContractName}`);
+    }
+    if (consensusContractName !== "PolygonPessimisticConsensus" && verifierType !== 0) {
+        throw new Error(`Verifier type should be 0 for ${consensusContractName}`);
+    }
+
+    // Grant role CREATE_ROLLUP_ROLE to deployer
+    const CREATE_ROLLUP_ROLE = ethers.id("CREATE_ROLLUP_ROLE");
+    if ((await rollupManagerContract.hasRole(CREATE_ROLLUP_ROLE, deployer.address)) == false)
+        await rollupManagerContract.grantRole(CREATE_ROLLUP_ROLE, deployer.address);
+
+    // Get rollup address deterministically
+    const nonce = await currentProvider.getTransactionCount(rollupManagerContract.target);
+    const createdRollupAddress = ethers.getCreateAddress({
+        from: rollupManagerContract.target as string,
+        nonce: nonce,
+    });
+
+    // Create new rollup
+    const txDeployRollup = await rollupManagerContract.createNewRollup(
+        createRollupParameters.rollupTypeId,
+        chainID,
+        rollupAdminAddress,
+        trustedSequencer,
+        createRollupParameters.gasTokenAddress,
+        trustedSequencerURL,
+        networkName
+    );
+    outputJson.gasTokenAddress = createRollupParameters.gasTokenAddress;
+
+    const receipt = (await txDeployRollup.wait()) as any;
+    const blockDeploymentRollup = await receipt?.getBlock();
+    const timestampReceipt = blockDeploymentRollup.timestamp;
+
+    console.log("#######################\n");
+    console.log(`Created new ${consensusContractName} Rollup:`, createdRollupAddress);
```

Comment:
Print also the rollupTypeID.
As mention in a comment before, this could be inconsistent.